### PR TITLE
feat: [#177066306] Handle and show details about payment outcome

### PR DESCRIPTION
**Pull request by Undermaken**
# :warning: THIS PR DEPENDS ON https://github.com/pagopa/io-app/pull/2837

## Short description
This PR handles the payment outcome code.
It could be a success (payment done) or a failure.
In both cases a dedicated screen will be shown to display all information about success or failure

### adds
- now an error toast is shown if the app can't retrive a PM session token
- in case of success the transactions will be reloaded 
- in case of success the payment id is stored (temporary disabled here https://github.com/pagopa/io-app/pull/2837)
- in case of success the screen displays also the email address where the PM sends the notification

https://user-images.githubusercontent.com/822471/109041775-ae544c00-76cf-11eb-9cfe-87fe67d6b363.mov

### CODE REVIEW

File: locales/en/index.yml [hunk 1]
```diff
@@ -1270,7 +1270,8 @@ wallet:
     payment:
       success:
         title: "Thank you for your payment!"
-        description: "We sent a receipt to your email address. You can also find it in your Wallet."
+        description1: "We sent a receipt to your email address"
```
Comment (thisisjp) [inline locales/en/index.yml:1273]:
```suggestion
        description1: "We sent a receipt to"
```

File: locales/it/index.yml [hunk 2]
```diff
@@ -1295,7 +1295,8 @@ wallet:
     payment:
       success:
         title: "Grazie, l’operazione è stata eseguita con successo!"
-        description: "Abbiamo inviato una ricevuta al tuo indirizzo email. Puoi trovarla anche nel tuo Portafoglio."
+        description1: "Abbiamo inviato una ricevuta al tuo indirizzo email"
```
Comment (thisisjp) [inline locales/it/index.yml:1298]:
```suggestion
        description1: "Abbiamo inviato una ricevuta a"
```

**Review by thisisjp — COMMENTED** [review]
@Undermaken I slightly changed the copy ;)

File: ts/screens/wallet/payment/TransactionSummaryScreen.tsx [hunk 3]
```diff
@@ -93,8 +92,6 @@ const NOTICE_ICON_SIZE = 24;
  */
 class TransactionSummaryScreen extends React.Component<Props> {
   public componentDidMount() {
-    // refresh wallet list (to avoid cache de-sync)
-    this.props.dispatchWalletsLoad();
```
Comment (debiff) [inline ts/screens/wallet/payment/TransactionSummaryScreen.tsx:97]: Why this operation is no more needed?

**Review by debiff — APPROVED** [review]

File: ts/screens/wallet/payment/TransactionSummaryScreen.tsx (hunk 3 above)
Comment (Undermaken) [reply ts/screens/wallet/payment/TransactionSummaryScreen.tsx:97]: I added this refresh in the previous PR https://github.com/pagopa/io-app/pull/2837/files
but I found some strange behaviors when the wallet is empty. So I restore it as it was

**Pull request merged**